'Add to library list' and 'Set as current library' actions on browsers items

sebjulliand opened the pull request:
### Changes
This PR adds a new `Set as current library` action, available from:
- Libraries in the `User Library List` view
- Single library Filters and .lib objects from the `Object Browser`

It adds the `Add to library list` action on:
- Single library Filters and .lib objects from the `Object Browser`

![image](https://github.com/codefori/vscode-ibmi/assets/11096890/fa9521c3-f1cb-4290-9bb3-b0159630b24d)

It also allows to run `Create source file` action on:
- Single library  Filters and .lib objects from the `Object Browser`

Finally, it removes `Add to library list...` from command palette when library list view is disabled

@chrjorgensen, I'll need your help for one translation 😄 

### How to test
- Try `Add to library list` action from:
  - A library from the `User Library List` view
  - A filter from the `Object Browser`
  - A .lib object from the `Object Browser`
  - The `User Library List` view's title
- Try `Set as current library` action from:
  - A library from the `User Library List` view
  - A filter from the `Object Browser`
  - A .lib object from the `Object Browser`

### Checklist
* [x] have tested my change

chrjorgensen reviewed (changes requested):
Nice enhancement, @sebjulliand ! :+1: 

I translated the new error message to Danish and changed a command title to title case.

It's working as expected - only one issue was found:
- filters having multiple libraries
  - Add to Library List returns error `Library is too long.`
  - Set to Current Library does nothing

In addition, the context menu item `Add to Library List...` does not prompt the user for any information and should not have `...` in the end... however, when it's run from the command Palette, it prompts the user and should have `...`.
AFAICT the only solution to this situation is two commands with a correct title - where one should only be for the title and should call the other for the action. Too much trouble?

sebjulliand commented:
You're right @chrjorgensen, as always!
I introduced a new type (`WithLibrary`) and modified the object browser's items context so the library related actions (including `Create source file`) while only appear on library nodes. This gives a better UX and a cleaner code overall.

I updated the description of the PR accordingly.

chrjorgensen commented on package.json:
```diff
@@ -1104,6 +1104,12 @@
+				"command": "code-for-ibmi.setCurrentLibrary",
+				"enablement": "code-for-ibmi:connected",
+				"title": "Set as Current Library",
+				"category": "IBM i"				
```
A few tabs appending the line...

chrjorgensen commented on src/typings.ts:
```diff
@@ -157,7 +157,9 @@ export interface WithPath {
 
-export interface Library extends WithPath { }
+export interface WithLibrary {
+  library:string
```
```suggestion
  library: string
```

sebjulliand commented:
@chrjorgensen formatting done!

chrjorgensen reviewed (approved):
@sebjulliand Works like a charm - and looks good! 🎉
